tests/foreman/ui/test_rhcloud_insights_vulnerability.py

```diff
@@ -0,0 +1,90 @@
+    # Upload insights data to Satellite
+    assert host.execute('insights-client').status == 0
+
+    # Add vulnerability for glibc package - CVE-2025-8058
+    assert host.execute('dnf downgrade -y glibc').status == 0
+
+
+@pytest.mark.e2e
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_list([10])
+@pytest.mark.parametrize('module_target_sat_insights', [False], ids=['local'], indirect=True)
+def test_rhcloud_insights_vulnerabilities_e2e(
+    request,
+    rhel_insights_vm,
+    rhcloud_manifest_org,
+    module_target_sat_insights,
+):
+    """Validate vulnerability data from local Insights,
+        verify results are displayed correctly for impacted host in Satellite.
+
+    :id: d952e83c-3faf-4299-a048-2eb6ccb8c9c3
+
+    :steps:
+        1. Enable local Insights/Red Hat Lightspeed on Satellite.
+        2. Import Manifest, Sync RHEL content and create AK with the content.
+        3. Register a VM using AK created.
+        4. Create vulnerabilities for detection by vulnerability engine and upload its data to Insights.
+        5. In Satellite UI, go to vulnerabilities tab on host details page and validate the data.
+
+    :expectedresults:
+        1. Local insights vulnerability engine detects the new vulnerabilities.
+        2. Vulnerability data is displayed correctly for impacted host in Satellite.
+
+    :verifies: SAT-30762
+    """
+    org_name = rhcloud_manifest_org.name
+    CVE_ID = 'CVE-2025-8058'
+
+    # Verify insights-client can update to latest version available from server
+    assert rhel_insights_vm.execute('insights-client --version').status == 0
+
+    # Manual VMaaS reposcan sync task
+    assert (
+        module_target_sat_insights.execute(
+            'curl --fail --cert /etc/foreman/client_cert.pem --key /etc/foreman/client_key.pem --cacert /etc/foreman/proxy_ca.pem -X PUT "https://localhost:24443/api/vmaas-reposcan/v1/sync"'
+        ).status
+        == 0
+    )
+
+    # Prepare a machine with vulnerability CVE-2025-8058 and upload data to Insights
+    create_insights_vulnerability(rhel_insights_vm)
+
+    with module_target_sat_insights.ui_session() as session:
+        session.organization.select(org_name=org_name)
+
+        status = session.host_new.get_host_statuses(rhel_insights_vm.hostname)
+        assert status['Red Hat Lightspeed']['Status'] == 'Reporting'
+
+        # Read the CVEs listed on Vulnerabiltities tab on host details page
+        vulnerabilities = session.host_new.get_vulnerabilities(rhel_insights_vm.hostname)
+        assert vulnerabilities[0]['CVE ID'] == CVE_ID
+
+        # Validate the affected host from vulnerability details and list pages
+        affected_host = session.cloudvulnerability.get_affected_hosts_by_cve(CVE_ID)
+        assert affected_host[0]['Name'] == rhel_insights_vm.hostname
+
+        # Validate the vulnerabilities tab on host details page from CVE list and CVE details
+        vulnerabilities = session.cloudvulnerability.validate_cve_to_host_flow(
```

Comment:
I didn't find the definition for `validate_cve_to_host_flow`. Where the method is defined?

Reply:
https://github.com/SatelliteQE/airgun/pull/1978/files#diff-f1d3fdb0e53f8332c9fae0628dff282b799d35d2a82edc2a495e22306a1529e2R44